### ISSUE TITLE
Run web-platform-tests without similar-origin REMOTE_HOST

### ISSUE THREAD

**Issue by asajeffrey** [servo/servo#15232]
At the moment, we're running the web-platform-tests with `ORIGINAL_HOST=web-platform.test` and `REMOTE_HOST=www1.web-platform.test`, which is similar-origin. This means that when we're running tests of cross-origin behaviour, we're only testing similar-origin. In particular, we're not running tests across script threads.

cc @jdm @jgraham

**Comment by jgraham** [comment]
Which tests exactly do you care about here? It seems like the easiest thing to do would be to switch either port or protocol rather than reserving another domain.

**Comment by asajeffrey** [comment]
@jgraham the ones which deal with cross-origin iframes, e.g. `/html/browsers/origin/cross-origin-objects/cross-origin-objects-exceptions.html`. It would be good enough to run with `ORIGINAL_HOST` being `localhost`, since then `REMOTE_HOST` would be set to `127.0.0.1`. Slightly annoyingly, the Servo mapping from eTLD to event loop is only done based on the host name, not the port or protocol.

**Comment by asajeffrey** [comment]
Running master on a hacked version of wpt that sets the remote to be `127.0.0.1`:
```
$ ./mach test-wpt --release
Running 8017 tests in web-platform-tests

... some intermittents ...

  ▶ Unexpected subtest result in /fetch/api/basic/mode-same-origin-worker.html:
  │ FAIL [expected PASS] Fetch http://127.0.0.1:8000/fetch/api/resources/top.txt with same-origin mode
  └   → assert_unreached: Should have rejected: undefined Reached unreachable code

  ▶ Unexpected subtest result in /fetch/api/basic/mode-same-origin.html:
  │ FAIL [expected PASS] Fetch http://127.0.0.1:8000/fetch/api/resources/top.txt with same-origin mode
  └   → assert_unreached: Should have rejected: undefined Reached unreachable code

Ran 8017 tests finished in 3879.0 seconds.
  • 8010 ran as expected. 1889 tests skipped.
  • 4 tests failed unexpectedly
  • 1 tests passed unexpectedly
  • 2 tests had unexpected subtest results

```

**Comment by asajeffrey** [comment]
IRC chat with @jdm and @KiChjang: http://logs.glob.uno/?c=mozilla%23servo&s=26+Jan+2017&e=26+Jan+2017#c600869

TL;DR edit `env_options` in `tests/wpt/harness/wptrunner/browsers/servo.py`. Setting `host=locahost` causes errors:
```
$ ./mach test-wpt /fetch/api/basic
...
Summary
=======

Ran 346 tests (35 parents, 311 subtests)
Expected results: 340
Unexpected results: 6 (FAIL: 6)

Unexpected Results
==================

/fetch/api/basic/mode-no-cors.html
----------------------------------
FAIL Fetch https://localhost:8443/fetch/api/resources/top.txt with no-cors mode
FAIL Fetch http://localhost:8001/fetch/api/resources/top.txt with no-cors mode
/fetch/api/basic/mode-no-cors-worker.html
-----------------------------------------
FAIL Fetch https://localhost:8443/fetch/api/resources/top.txt with no-cors mode
FAIL Fetch http://localhost:8001/fetch/api/resources/top.txt with no-cors mode
/fetch/api/basic/mode-same-origin-worker.html
---------------------------------------------
FAIL Fetch http://127.0.0.1:8000/fetch/api/resources/top.txt with same-origin mode
/fetch/api/basic/mode-same-origin.html
--------------------------------------
FAIL Fetch http://127.0.0.1:8000/fetch/api/resources/top.txt with same-origin mode
```
These are errors caused by #15247.

**Comment by asajeffrey** [comment]
Submitted wptrunner PR https://github.com/w3c/wptrunner/pull/231 to allow running with `--external-host=localhost`, which produces the same `FAIL`s as above.

**Comment by asajeffrey** [comment]
We should probably update the CI to run some of the wpt tests with `--external-host=localhost`, especially the ones which test same-origin policy, e.g. `/fetch` and `/html/browsers`. cc @aneeshusa

**Comment by asajeffrey** [comment]
The long-term version of how to do this is for wpt to have dissimilar-origin domains. For the moment I just added some dissimilar-origin tests (with the host hard-wired to be `127.0.0.1`) in `/mozilla`.
